### PR TITLE
[WIP][Kernel] Set the built-in reduce result of zero-degree nodes to 0 in C

### PR DESCRIPTION
**Pull request by BarclayII**
## Description
We have decided to set the built-in reduce result of zero-degree nodes to 0 regardless of the reducer.  This is currently done in Python.  I'm wondering whether we can lower the logic to C which should make things faster.

For sum and mean reducers, we do nothing.

For max and min reducers, we simply replace all -infinity and infinity entries to 0.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

File: src/array/cpu/spmm.h [hunk 1]
```diff
@@ -104,8 +104,10 @@ void SpMMSumCoo(
       const DType* lhs_off = Op::use_lhs? X + rid * lhs_dim + lhs_add : nullptr;
       const DType* rhs_off = Op::use_rhs? W + eid * rhs_dim + rhs_add : nullptr;
       const DType val = Op::Call(lhs_off, rhs_off);
+      if (val != 0) {
```
Comment (yzh119) [inline src/array/cpu/spmm.h:107]: Why do we need this judgement?

Comment (BarclayII) [reply src/array/cpu/spmm.h:107]: See #1527

File: src/array/cpu/spmm.h [hunk 2]
```diff
@@ -315,7 +328,10 @@ template <typename DType> constexpr bool CopyRhs<DType>::use_rhs;
 //////////////////////////////// Reduce operators on CPU ////////////////////////////////
 template <typename DType>
 struct Max {
-  static constexpr DType zero = std::numeric_limits<DType>::lowest();
+  static constexpr DType zero = (
+      std::numeric_limits<DType>::has_infinity ?
+      -std::numeric_limits<DType>::infinity() :
+      std::numeric_limits<DType>::lowest());
```
Comment (yzh119) [inline src/array/cpu/spmm.h:334]: Why don't we just use `lowest` and `max`?

Comment (BarclayII) [reply src/array/cpu/spmm.h]: Now I'm only using infinity.

**Pull request merged**